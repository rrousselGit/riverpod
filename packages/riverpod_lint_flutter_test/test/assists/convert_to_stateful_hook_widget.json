[{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":221,"length":0,"replacement":"  @override\n    State<Stateless> createState() => _StatelessHookState();\n  }\n\n  class _StatelessHookState extends State<Stateless> {\n  "},{"offset":168,"length":15,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":404,"length":0,"replacement":"  @override\n    State<StatelessWithComma> createState() => _StatelessWithCommaHookState();\n  }\n\n  class _StatelessWithCommaHookState extends State<StatelessWithComma> {\n  "},{"offset":342,"length":15,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":563,"length":0,"replacement":"  @override\n    State<Hook> createState() => _HookHookState();\n  }\n\n  class _HookHookState extends State<Hook> {\n  "},{"offset":520,"length":10,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":787,"length":23,"replacement":""},{"offset":737,"length":0,"replacement":"  @override\n    State<HookConsumer> createState() => _HookConsumerHookState();\n  }\n\n  class _HookConsumerHookState extends State<HookConsumer> {\n  "},{"offset":678,"length":18,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":1022,"length":15,"replacement":"State<Stateful>"},{"offset":939,"length":15,"replacement":"State<Stateful>"},{"offset":876,"length":14,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":1389,"length":26,"replacement":"State<ExplicitCreateState>"},{"offset":1203,"length":14,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":2045,"length":31,"replacement":"State<ConsumerStateful>"},{"offset":1930,"length":31,"replacement":"State<ConsumerStateful>"},{"offset":1851,"length":22,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":2424,"length":35,"replacement":"State<HookConsumerStateful>"},{"offset":2291,"length":35,"replacement":"State<HookConsumerStateful>"},{"offset":2204,"length":26,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}},{"priority":97,"change":{"message":"Convert to StatefulHookWidget","edits":[{"fileStamp":0,"edits":[{"offset":2676,"length":23,"replacement":""},{"offset":2626,"length":0,"replacement":"  @override\n    State<Consumer> createState() => _ConsumerHookState();\n  }\n\n  class _ConsumerHookState extends State<Consumer> {\n  "},{"offset":2575,"length":14,"replacement":"StatefulHookWidget"}]}],"linkedEditGroups":[]}}]