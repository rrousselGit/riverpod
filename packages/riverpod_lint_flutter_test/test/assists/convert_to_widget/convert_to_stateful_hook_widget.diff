Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:5`:
```
import 'package:flutter_hooks/flutter_hooks.dart';

- class Stateless extends StatelessWidget {
-   const Stateless({super.key});
- 
-   @override
+ class Stateless extends StatefulHookWidget {
+   const Stateless({super.key});
+ 
+   @override
+   State<Stateless> createState() => _StatelessState();
+ }
+ 
+ class _StatelessState extends State<Stateless> {
@override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:14`:
```
}

- class StatelessWithComma extends StatelessWidget {
-   const StatelessWithComma({super.key});
- 
-   @override
+ class StatelessWithComma extends StatefulHookWidget {
+   const StatelessWithComma({super.key});
+ 
+   @override
+   State<StatelessWithComma> createState() => _StatelessWithCommaState();
+ }
+ 
+ class _StatelessWithCommaState extends State<StatelessWithComma> {
@override
  Widget build(
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:25`:
```
}

- class Hook extends HookWidget {
-   const Hook({super.key});
- 
-   @override
+ class Hook extends StatefulHookWidget {
+   const Hook({super.key});
+ 
+   @override
+   State<Hook> createState() => _HookState();
+ }
+ 
+ class _HookState extends State<Hook> {
@override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:34`:
```
}

- class HookConsumer extends HookConsumerWidget {
-   const HookConsumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class HookConsumer extends StatefulHookWidget {
+   const HookConsumer({super.key});
+ 
+   @override
+   State<HookConsumer> createState() => _HookConsumerState();
+ }
+ 
+ class _HookConsumerState extends State<HookConsumer> {
+ @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to StatefulHookWidget`
Priority: 28
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:46`:
```
}

- class Stateful extends StatefulWidget {
+ class Stateful extends StatefulHookWidget {
  const Stateful({super.key});

```
---
Message: `Convert to StatefulHookWidget`
Priority: 28
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:62`:
```
}

- class ExplicitCreateState extends StatefulWidget {
+ class ExplicitCreateState extends StatefulHookWidget {
  const ExplicitCreateState({super.key});

```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:92`:
```
}

- class ConsumerStateful extends ConsumerStatefulWidget {
-   const ConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
- }
- 
- class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
+ class ConsumerStateful extends StatefulHookWidget {
+   const ConsumerStateful({super.key});
+ 
+   @override
+   State<ConsumerStateful> createState() => _ConsumerStatefulState();
+ }
+ 
+ class _ConsumerStatefulState extends State<ConsumerStateful> {
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:106`:
```
}

- class HookConsumerStateful extends StatefulHookConsumerWidget {
-   const HookConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<HookConsumerStateful> createState() =>
-       _HookConsumerStatefulState();
- }
- 
- class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
+ class HookConsumerStateful extends StatefulHookWidget {
+   const HookConsumerStateful({super.key});
+ 
+   @override
+   State<HookConsumerStateful> createState() =>
+       _HookConsumerStatefulState();
+ }
+ 
+ class _HookConsumerStatefulState extends State<HookConsumerStateful> {
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:121`:
```
}

- class Consumer extends ConsumerWidget {
-   const Consumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class Consumer extends StatefulHookWidget {
+   const Consumer({super.key});
+ 
+   @override
+   State<Consumer> createState() => _ConsumerState();
+ }
+ 
+ class _ConsumerState extends State<Consumer> {
+ @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:133`:
```
}

- class StatelessWithField extends StatelessWidget {
-   const StatelessWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(BuildContext context) {
-     return Column(
-       children: [
-         Text('$field'),
-         Text('$staticField'),
+ class StatelessWithField extends StatefulHookWidget {
+   const StatelessWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   State<StatelessWithField> createState() => _StatelessWithFieldState();
+ }
+ 
+ class _StatelessWithFieldState extends State<StatelessWithField> {
+ @override
+   Widget build(BuildContext context) {
+     return Column(
+       children: [
+         Text('${widget.field}'),
+         Text('${StatelessWithField.staticField}'),
      ],
    );
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:153`:
```
}

- class HookConsumerWithField extends HookConsumerWidget {
-   const HookConsumerWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
-     return Column(
-       children: [
-         Text('$field'),
-         Text('$staticField'),
+ class HookConsumerWithField extends StatefulHookWidget {
+   const HookConsumerWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   State<HookConsumerWithField> createState() => _HookConsumerWithFieldState();
+ }
+ 
+ class _HookConsumerWithFieldState extends State<HookConsumerWithField> {
+ @override
+   Widget build(
+     BuildContext context) {
+     return Column(
+       children: [
+         Text('${widget.field}'),
+         Text('${HookConsumerWithField.staticField}'),
      ],
    );
```
---
Message: `Convert to StatefulHookWidget`
Priority: 32
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:180`:
```
}

- class ConsumerStatefulWithField extends ConsumerStatefulWidget {
-   const ConsumerStatefulWithField({
-     super.key,
-     required this.field,
-     required this.foo,
-   });
- 
-   final int field;
-   final FooClass foo;
-   static final int staticField = _constantNumber;
- 
-   @override
-   ConsumerState<ConsumerStatefulWithField> createState() =>
-       _ConsumerStatefulWithFieldState();
- }
- 
- const _constantNumber = 42;
- 
- class _ConsumerStatefulWithFieldState
-     extends ConsumerState<ConsumerStatefulWithField> {
+ class ConsumerStatefulWithField extends StatefulHookWidget {
+   const ConsumerStatefulWithField({
+     super.key,
+     required this.field,
+     required this.foo,
+   });
+ 
+   final int field;
+   final FooClass foo;
+   static final int staticField = _constantNumber;
+ 
+   @override
+   State<ConsumerStatefulWithField> createState() =>
+       _ConsumerStatefulWithFieldState();
+ }
+ 
+ const _constantNumber = 42;
+ 
+ class _ConsumerStatefulWithFieldState
+     extends State<ConsumerStatefulWithField> {
  void printFoo() {
    print(widget.foo);
```
---
