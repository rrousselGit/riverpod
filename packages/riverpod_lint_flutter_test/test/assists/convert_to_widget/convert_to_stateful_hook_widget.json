[
  {
    "priority": 32,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 221,
              "length": 0,
              "replacement": "@override\n  State<Stateless> createState() => _StatelessState();\n}\n\nclass _StatelessState extends State<Stateless> {\n"
            },
            {
              "offset": 168,
              "length": 15,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 32,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 404,
              "length": 0,
              "replacement": "@override\n  State<StatelessWithComma> createState() => _StatelessWithCommaState();\n}\n\nclass _StatelessWithCommaState extends State<StatelessWithComma> {\n"
            },
            {
              "offset": 342,
              "length": 15,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 32,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 563,
              "length": 0,
              "replacement": "@override\n  State<Hook> createState() => _HookState();\n}\n\nclass _HookState extends State<Hook> {\n"
            },
            {
              "offset": 520,
              "length": 10,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 32,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 787,
              "length": 23,
              "replacement": ""
            },
            {
              "offset": 737,
              "length": 0,
              "replacement": "@override\n  State<HookConsumer> createState() => _HookConsumerState();\n}\n\nclass _HookConsumerState extends State<HookConsumer> {\n"
            },
            {
              "offset": 678,
              "length": 18,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 28,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 1022,
              "length": 15,
              "replacement": "State<Stateful>"
            },
            {
              "offset": 939,
              "length": 15,
              "replacement": "State<Stateful>"
            },
            {
              "offset": 876,
              "length": 14,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 28,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 1389,
              "length": 26,
              "replacement": "State<ExplicitCreateState>"
            },
            {
              "offset": 1203,
              "length": 14,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 32,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 2045,
              "length": 31,
              "replacement": "State<ConsumerStateful>"
            },
            {
              "offset": 1930,
              "length": 31,
              "replacement": "State<ConsumerStateful>"
            },
            {
              "offset": 1851,
              "length": 22,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 32,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 2424,
              "length": 35,
              "replacement": "State<HookConsumerStateful>"
            },
            {
              "offset": 2291,
              "length": 35,
              "replacement": "State<HookConsumerStateful>"
            },
            {
              "offset": 2204,
              "length": 26,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  },
  {
    "priority": 32,
    "change": {
      "message": "Convert to StatefulHookWidget",
      "edits": [
        {
          "fileStamp": 0,
          "edits": [
            {
              "offset": 2676,
              "length": 23,
              "replacement": ""
            },
            {
              "offset": 2626,
              "length": 0,
              "replacement": "@override\n  State<Consumer> createState() => _ConsumerState();\n}\n\nclass _ConsumerState extends State<Consumer> {\n"
            },
            {
              "offset": 2575,
              "length": 14,
              "replacement": "StatefulHookWidget"
            }
          ]
        }
      ],
      "linkedEditGroups": []
    }
  }
]