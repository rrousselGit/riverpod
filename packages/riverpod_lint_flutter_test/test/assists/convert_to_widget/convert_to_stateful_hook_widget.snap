Message: `Convert to StatefulHookWidget`
<<<< start: 173 -- end: 220
lessWidget {
  const Stateless({super.key});

 
==== start: 173 -- end: 340
fulHookWidget {
  const Stateless({super.key});

  @override
  State<Stateless> createState() => _StatelessState();
}

class _StatelessState extends State<Stateless> {
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 347 -- end: 403
lessWidget {
  const StatelessWithComma({super.key});

 
==== start: 347 -- end: 559
fulHookWidget {
  const StatelessWithComma({super.key});

  @override
  State<StatelessWithComma> createState() => _StatelessWithCommaState();
}

class _StatelessWithCommaState extends State<StatelessWithComma> {
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 520 -- end: 562
HookWidget {
  const Hook({super.key});

 
==== start: 520 -- end: 667
StatefulHookWidget {
  const Hook({super.key});

  @override
  State<Hook> createState() => _HookState();
}

class _HookState extends State<Hook> {
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 678 -- end: 809
HookConsumerWidget {
  const HookConsumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 678 -- end: 915
StatefulHookWidget {
  const HookConsumer({super.key});

  @override
  State<HookConsumer> createState() => _HookConsumerState();
}

class _HookConsumerState extends State<HookConsumer> {
@override
  Widget build(
    BuildContext contex
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 884 -- end: 884
==== start: 884 -- end: 888
Hook
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 1211 -- end: 1211
==== start: 1211 -- end: 1215
Hook
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 1851 -- end: 2052
ConsumerStatefulWidget {
  const ConsumerStateful({super.key});

  @override
  ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends Consume
==== start: 1851 -- end: 2032
StatefulHookWidget {
  const ConsumerStateful({super.key});

  @override
  State<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 2216 -- end: 2431
ConsumerWidget {
  const HookConsumerStateful({super.key});

  @override
  ConsumerState<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends Consume
==== start: 2216 -- end: 2407
Widget {
  const HookConsumerStateful({super.key});

  @override
  State<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends
>>>>
Message: `Convert to StatefulHookWidget`
<<<< start: 2575 -- end: 2698
ConsumerWidget {
  const Consumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 2575 -- end: 2792
StatefulHookWidget {
  const Consumer({super.key});

  @override
  State<Consumer> createState() => _ConsumerState();
}

class _ConsumerState extends State<Consumer> {
@override
  Widget build(
    BuildContext contex
>>>>
